x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality/asset_criticality_selector.tsx

```diff
@@ -70,77 +170,35 @@ const AssetCriticalityComponent: React.FC<Props> = ({ entity }) => {
         }}
         data-test-subj="asset-criticality-selector"
       >
-        {criticality.query.isLoading || criticality.mutation.isLoading ? (
-          <EuiLoadingSpinner size="s" />
-        ) : (
-          <EuiFlexGroup
-            direction="row"
-            alignItems="center"
-            justifyContent="spaceBetween"
-            wrap={false}
-          >
-            <EuiFlexItem>
-              <EuiText size="s">
-                <AssetCriticalityBadgeAllowMissing
-                  criticalityLevel={criticality.query.data?.criticality_level}
-                  dataTestSubj="asset-criticality-level"
-                />
-              </EuiText>
-            </EuiFlexItem>
-            {criticality.privileges.data?.has_write_permissions && (
-              <EuiFlexItem css={{ flexGrow: 'unset' }}>
-                <EuiButtonEmpty
-                  data-test-subj="asset-criticality-change-btn"
-                  iconType="arrowStart"
-                  iconSide="left"
-                  flush="right"
-                  onClick={() => modal.toggle(true)}
-                >
-                  {criticality.status === 'update' ? (
-                    <FormattedMessage
-                      id="xpack.securitySolution.entityAnalytics.assetCriticality.changeButton"
-                      defaultMessage="Change"
-                    />
-                  ) : (
-                    <FormattedMessage
-                      id="xpack.securitySolution.entityAnalytics.assetCriticality.createButton"
-                      defaultMessage="Create"
-                    />
-                  )}
-                </EuiButtonEmpty>
-              </EuiFlexItem>
-            )}
-          </EuiFlexGroup>
-        )}
+        <AssetCriticalitySelector criticality={criticality} entity={entity} />
       </EuiAccordion>
       <EuiHorizontalRule />
-      {modal.visible ? (
-        <AssetCriticalityModal entity={entity} criticality={criticality} modal={modal} />
-      ) : null}
     </>
   );
 };
 
 interface ModalProps {
   criticality: State;
-  modal: ModalState;
+  onClose: () => void;
   entity: Entity;
 }
-const AssetCriticalityModal: React.FC<ModalProps> = ({ criticality, modal, entity }) => {
+
+const AssetCriticalityModal: React.FC<ModalProps> = ({ criticality, entity, onClose }) => {
+  const basicSelectId = useGeneratedHtmlId({ prefix: 'basicSelect' });
   const [value, setNewValue] = useState<CriticalityLevel>(
     criticality.query.data?.criticality_level ?? 'normal'
   );
 
   return (
-    <EuiModal onClose={() => modal.toggle(false)}>
+    <EuiModal onClose={onClose}>
```

Comment:
although this is more of a nitpick, it's kinda the same here. Is there a reason why we should change these handlers and props?
I feel like the previous version was a bit more semantic and readable.